release-content/0.15/release-notes/14791_Required_Components.md

```diff
@@ -1,4 +1,421 @@
+
+### What are they?
+
+**Required Components** enable developers to define which components a given component needs:
+
+```rust
+#[derive(Component, Default)]
+#[require(Team, Sprite)]
+struct Player {
+    name: String,
+}
+```
+
+When the `Player` component is inserted, its **Required Components** _and the components required by those components_ are automatically inserted as well!
+
+```rust
+commands.spawn(Player::default());
+```
+
+The code above automatically inserts `Team` and `Sprite`. `Sprite` requires `Transform` and `Visibility`, so those are automatically inserted as well. Likewise `Transform` requires `GlobalTransform` and `Visibility` requires `InheritedVisibility` and `ViewVisibility`.
+
+This code produces the same result as the `PlayerBundle` code in the previous section:
+
+```rust
+commands.spawn((
+    Player {
+        name: "hello".into(),
+        ..default()
+    },
+    Team::Blue,
+))
+```
+
+Much better right? The `Player` type is easier and less error prone to define, and spawning it takes less typing and is easier to read.
+
+### Efficiency
+
+We've implemented **Required Components** in a way that makes them effectively "free":
+
+1. Required Components are only initialized and inserted if the caller did not insert them manually. No redundancy!
+2. Required Components are inserted alongside the normal components, meaning (for you ECS nerds out there) there are no additional archetype changes or table moves. From this perspective, the Required Components version of the `Player` example is identical to the `PlayerBundle` approach, which manually defines all of the components up front.
+3. Required Components are cached on the archetype graph, meaning computing what required components are necessary for a given type of insert only happens once.
+
+### Component Initialization
+
+By default, **Required Components** will use the `Default` impl for the component (and fail to compile if one does not exist):
+
+```rust
+#[derive(Component)]
+#[require(Team)] // Team::Red is the default value
+struct Player;
+
+#[derive(Default)]
+enum Team {
+    #[default]
+    Red,
+    Blue,
+}
+```
+
+This can be overridden by passing in a function that returns the component:
+
+```rust
+#[derive(Component)]
+#[require(Team(blue_team)]
+struct Player;
+
+fn blue_team() -> Team {
+    Team::Blue
+}
+```
+
+To save space, you can also pass a closure to the require directly:
+
+```rust
+#[derive(Component)]
+#[require(Team(|| Team::Blue))]
+struct Player;
+```
+
+### Isn't this a bit like inheritance?
+
+**Required Components** _can_ be considered a form of inheritance. But it is notably _not_ traditional object-oriented inheritance. Instead it is "inheritance by composition". A `Button` widget can (and should) require `Node` to make it a "UI node". In a way, a `Button` "is a" `Node`. But unlike traditional inheritance:
```

Comment:
I think it would be clearer to describe this as a "has-a" relationship, and compare "is-a". To my ears, attempting to describe required components as inheritance rather than composition muddies the waters a little. If anything, I suppose, it's a form of _mandatory composition_?

Reply:
Good call. Just clarified this